### DISCUSSION TITLE
How make a map

### DISCUSSION THREAD

**Discussion by realroot2185** [FyroxEngine/Fyrox#622] [Q&A]
For example how make a terrain that will be the ground?
I suppose that that's how you do it.

Or a see.
Are there scene with a see that one can use?

I added a bush model but when I start the game it goes straight in the sky.

What type should be the root of the scene?

**Comment by mrDIMAS** [comment]
Create a rigid body with a collider, that has Heightmap shape. Set terrain as a geometry source for the shape, set rigid body type to Static. More info here - https://fyrox-book.github.io/scene/terrain_node.html#physics and https://fyrox-book.github.io/physics/rigid_body.html. In your case it seems that all your bodies are dynamic, thus all of them are constantly falling.

**Reply by realroot2185** [reply]
>Then attach the terrain to the rigid body. Keep in mind that terrain's origin differs from Heightmap rigid body, so you need to offset the terrain to match its physical representation. Enable physics visualization in editor settings to see physical shapes and move terrain.

How do you check that? Terrain and rigid body have the same y. 
Visually looks hard to spot the small amount of difference that seems to exist.
Since I see the paladin above the ground a little.

When I launch the game the paladin cannot move or the terrain rotate, under the pressure of the paladin it seems.
Without changing anything in the code/editor.

I set rigid body type to Static. Do I need to change mobility too?

**Reply by mrDIMAS** [reply]
Your character's rigid body must be Dynamic, also make sure to uncheck `Can Sleep` flag.

**Reply by realroot2185** [reply]
>I set rigid body type to Static. Do I need to change mobility too?

I mean for the root node rigid body, whose children is the terrain.

I now uncheck `Can Sleep` flag for character's rigid body, that is Dynamic.
It is a before.
